examples/secure_cloud_function_internal_server/main.tf

```diff
@@ -0,0 +1,234 @@
+#  * limitations under the License.
+#  */
+
+locals {
+  location           = "us-west1"
+  region             = "us-west1"
+  zone               = "us-west1-b"
+  repository_name    = "rep-secure-cloud-function"
+  network_ip         = "10.0.0.3"
+  webserver_instance = "webserver"
+  subnet_ip          = "10.0.0.0/28"
+}
+resource "random_id" "random_folder_suffix" {
+  byte_length = 2
+}
+
+module "secure_harness" {
+  source  = "GoogleCloudPlatform/cloud-run/google//modules/secure-serverless-harness"
+  version = "~> 0.8"
+
+  billing_account                             = var.billing_account
+  security_project_name                       = "prj-security"
+  network_project_name                        = "prj-restricted-shared"
+  serverless_project_names                    = ["prj-secure-cloud-function"]
+  org_id                                      = var.org_id
+  parent_folder_id                            = var.folder_id
+  serverless_folder_suffix                    = random_id.random_folder_suffix.hex
+  region                                      = local.region
+  location                                    = local.location
+  vpc_name                                    = "vpc-secure-cloud-function"
+  subnet_ip                                   = local.subnet_ip
+  private_service_connect_ip                  = "10.3.0.5"
+  create_access_context_manager_access_policy = var.create_access_context_manager_access_policy
+  access_context_manager_policy_id            = var.access_context_manager_policy_id
+  access_level_members                        = distinct(concat(var.access_level_members, ["serviceAccount:${var.terraform_service_account}"]))
+  key_name                                    = "key-secure-artifact-registry"
+  keyring_name                                = "krg-secure-artifact-registry"
+  prevent_destroy                             = false
+  artifact_registry_repository_name           = local.repository_name
+  egress_policies                             = var.egress_policies
+  ingress_policies                            = var.ingress_policies
+  serverless_type                             = "CLOUD_FUNCTION"
+  use_shared_vpc                              = true
+  time_to_wait_vpc_sc_propagation             = "600s"
+
+  service_account_project_roles = {
+    "prj-secure-cloud-function" = [
+      "roles/eventarc.eventReceiver",
+      "roles/viewer",
+      "roles/compute.networkViewer",
+      "roles/run.invoker"
+    ]
+  }
+
+  network_project_extra_apis = [
+    "certificatemanager.googleapis.com",
+    "networkservices.googleapis.com",
+    "networksecurity.googleapis.com"
+  ]
+
+  serverless_project_extra_apis = {
+    "prj-secure-cloud-function" = [
+      "networksecurity.googleapis.com"
+    ]
+  }
+}
+
+resource "time_sleep" "wait_swp_and_certificate_to_destroy" {
+  destroy_duration = "5m"
+
+  depends_on = [
+    module.secure_harness
```

Comment:
I'm confuse with this depends_on.

shouldn't depends on  null_resource.generate_certificate" and the harness depends on this time_sleep?

Reply:
It's not needed, I'll remove!